LiteCore/Storage/SQLiteKeyStore.cc

```diff
@@ -522,10 +522,12 @@ namespace litecore {
 
     // Adds the 'expiration' column to the table.
     void SQLiteKeyStore::addExpiration() {
-        if ( mayHaveExpiration() ) return;
-        db()._logVerbose("Adding the `expiration` column & index to kv_%s", name().c_str());
-        db().execWithLock(subst("ALTER TABLE kv_@ ADD COLUMN expiration INTEGER; "
-                                "CREATE INDEX \"kv_@_expiration\" ON kv_@ (expiration) WHERE expiration not null"));
+        db().withFileLock([=]() {
```

Comment:
First of all, by moving the check (mayHaveExpiration) inside the transaction, it should fix the concurrent calling from two db connections. 
Another option is to leave the read-only transaction, mayHaveExpiration, alone, and include db().execWithLock call in a try state, check the error whether it's because duplicate. If so, it's all right. This will save us not having to open a transaction every time addExpiration is called.

Reply:
The particular error (duplicate column) does not have a particular code, the code is `SQLITE_ERROR`. So we would have to check the string, which is not good.